### ISSUE TITLE
can we have a different onPress for different piechart data with animation onPress of some selected data

### ISSUE THREAD

**Issue by siddhi-unthinkable-co** [Abhinandan-Kushwaha/react-native-gifted-charts#208]
**Describe the bug**
Prop to make different datas of Pie chart clickable, such that when we click some specific data it should shown up or animated like it's somewhat lifted outwards.

**To Reproduce**
1. Go to pi chart
2. Click on some data from it
3. Clicked data should be animated like it's lifted upward

**Expected behavior**
Clickable datas on the pie chart and animations to be added on the onPress of datas, in mobile application.

**Screenshots**

 - Device: all iPhone and android devices
 - Version of the library being used [1.2.18]

<img width="190" alt="Screenshot 2022-05-26 at 6 04 50 PM" src="https://user-images.githubusercontent.com/106315351/170491858-c3903add-d5e4-4baa-8ba9-763bfd09a28c.png">

**Comment by Abhinandan-Kushwaha** [comment]
Hi @siddhi-unthinkable-co 👋
Thanks for requesting the feature. It is now available in version `1.2.27`🎉

To focus the pie sections on press, just pass the prop `focusOnPress`.

When a pie section is focused, its radius is increased by 10% of the chart's radius. The increase in radius of the focused section can be controlled using the `extraRadiusForFocused` prop. Also, by default the focus gets toggled on consecutive presses of the same pie section. This can be stopped by setting the prop `toggleFocusOnPress` to `false` (its default value is true).

I hope it helps. Cheers🍺

**Comment by Abhinandan-Kushwaha** [comment]
Here's an example-
![pieFocus](https://user-images.githubusercontent.com/20596944/170603008-1a5628f0-61fa-4e5b-b9dd-d673f9ae21d3.gif)

The code for the above chart is-
```js
const pieData = [
  {value: 54, color: '#177AD5'},
  {value: 40, color: '#79D2DE'},
  {value: 20, color: '#ED6665'},
];

return(
  <PieChart
    data={pieData}
    showText
    textColor="black"
    radius={150}
    textSize={20}
    focusOnPress
    showValuesAsLabels
    showTextBackground
    textBackgroundRadius={26}
  />
);
```

**Comment by Abhinandan-Kushwaha** [comment]
Different `onPress` for different pie sections can be given in the data data array.

**Comment by siddhi-unthinkable-co** [comment]
https://user-images.githubusercontent.com/106315351/170667373-5f6ce261-9fe4-41ab-b287-9450c28be8dd.mov

This focusOnPress is creating some distortion in the UI

**Comment by Abhinandan-Kushwaha** [comment]
Hi @siddhi-unthinkable-co 👋
I just released a new version of the library `1.2.29`.
Please try this version. It should solve the UI issue. If it persists, please comment here with your code snippet.
